### PR TITLE
Rework of user-guide/scripting/lua; Issue #671

### PR DESCRIPTION
**Pull request by FiniteStateGit**
Fixed formatting, file structure and updated images for the user-guide/scripting/lua directory.

I added 'Basic Structure of a Component Entity Lua Script' from the Lumberyard documentation, without any reference to `IsMaster`. 

Rewrote the Debugging with the Lua Editor tutorial; resolves https://github.com/o3de/o3de.org/issues/671

Resolves https://github.com/o3de/o3de.org/issues/741

### CODE REVIEW

File: content/docs/user-guide/scripting/lua/lua-editor.md [hunk 1]
```diff
@@ -1,6 +1,70 @@
...
+| Redo | Ctrl+Y |
+| Replace | Ctrl+R |
+| Replace in open files | Ctrl+Shift+R |
+| Select all | Ctrl+A |
+|  Select to brace¹  | Ctrl+Shift+\] |
+| Transpose lines down | Ctrl+Shift+Down Arrow |
+| Transpose lines up | Ctrl+Shift+Up Arrow |
+| Uncomment selected block | Ctrl+Shift+K |
+| Undo | Ctrl+Z |
+| Unfold source functions | Alt+Shift+0 |
```
Comment (micronAMZN) [inline content/docs/user-guide/scripting/lua/lua-editor.md:39]: The style guide calls for bold input combinations. I can see an argument for not using bold as the table already highlights them, but for consistency... Also, is the Select to Brace hotkey correct?

```suggestion
| **Action** | **Keyboard Shortcut** |
| --- | --- |
| Comment selected block | **Ctrl+K** |
| Copy | **Ctrl+C** |
| Cut | **Ctrl+X** |
| Find | **Ctrl+F** |
| Find in open files | **Ctrl+Shift+F** |
| Find next | **F3** |
| Fold source functions | **Alt+0** |
| Go to line | **Ctrl+G** |
| Paste | **Ctrl+V** |
| Quick find local | **Ctrl+F3** |
| Quick find local reverse | **Ctrl+Shift+F3** |
| Redo | **Ctrl+Y** |
| Replace | **Ctrl+R** |
| Replace in open files | **Ctrl+Shift+R** |
| Select all | **Ctrl+A** |
| Select to brace¹ | **Ctrl+Shift+\]** |
| Transpose lines down | **Ctrl+Shift+Down Arrow** |
| Transpose lines up | **Ctrl+Shift+Up Arrow** |
| Uncomment selected block | **Ctrl+Shift+K** |
| Undo | **Ctrl+Z** |
| Unfold source functions | **Alt+Shift+0** |
```

File: content/docs/user-guide/scripting/lua/basic-lua-script.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+title: Basic Structure of a Lua Script
+toc: true
+weight: 200
+---
+
+## Script Table
+
+Scripts to be used as components contain a table (referred to as the **Script Table**), which provides the functionality for the script. In Lua, this table is treated like a class. The script table generally consists of the following:
+
++ An optional **Properties** table within the script table. The **Properties** table provides an interface that you can use to customize the script behavior from the editor.
```
Comment (micronAMZN) [inline content/docs/user-guide/scripting/lua/basic-lua-script.md:13]: Feels like we should go with italics here.

```suggestion
+ An optional **Properties Table** within the Script Table. The Properties Table provides an interface that you can use to customize the script behavior from the editor.
```

File: content/docs/user-guide/scripting/lua/debugging-tutorial.md [hunk 3]
```diff
@@ -1,171 +1,144 @@
...
+    function ConstantRotation:OnDeactivate()
+        self.tickBusHandler:Disconnect()
+    end
+
+    return ConstantRotation
+    ```
+    
+1. **Save** the script as `ConstantRotation.lua` in your project's directory.
+
+1. **Close** the Lua Editor.
```
Comment (micronAMZN) [inline content/docs/user-guide/scripting/lua/debugging-tutorial.md:55]: Although these tool names are a bit generic, we are treating them as proper nouns.

```suggestion
1. **Close** Lua Editor.
```

File: content/docs/user-guide/scripting/lua/debugging-tutorial.md [hunk 4]
```diff
@@ -1,171 +1,144 @@
...
+
+1. In the **Entity Outliner**, **left-click** an entity to add a **Lua Script** component to.
```
Comment (micronAMZN) [inline content/docs/user-guide/scripting/lua/debugging-tutorial.md:57]:
```suggestion
1. In **Entity Outliner**, **left-click** an entity to add a **Lua Script** component to.
```

File: content/docs/user-guide/scripting/lua/debugging-tutorial.md [hunk 5]
```diff
@@ -1,171 +1,144 @@
...
 
 1. In **Entity Inspector**, click **Add Component**, and then choose **Scripting**, **Lua Script**.
 
 1. In the **Entity Inspector** window, locate the **Lua Script** component, and then click the **Pick Lua Script** button next to the empty **Script** field.
```
Comment (micronAMZN) [inline content/docs/user-guide/scripting/lua/debugging-tutorial.md:61]:
```suggestion
1. In **Entity Inspector**, locate the **Lua Script** component, and then click the **Pick Lua Script** button next to the empty **Script** field.
```

File: content/docs/user-guide/scripting/lua/debugging-tutorial.md [hunk 6]
```diff
@@ -1,171 +1,144 @@
...
 
 ## Tutorial: Using Lua Editor for Debugging with O3DE Editor 
 
-{{< todo issue="https://github.com/o3de/o3de.org/issues/671" >}}
-Update this Lua debugging tutorial with newer instructions that do not require the Samples Project.
-{{< /todo >}}
+This tutorial shows you how to use O3DE Lua Editor to perform debugging operations on a sample script.
 
-This tutorial shows you how to use O3DE Editor to create a sample level in the **SamplesProject** project with a component entity that contains a Lua script component. You will learn how to open the script in Lua Editor and perform some sample debugging steps on the script.
+1. Open the **Lua Editor** from the **Tools** menu.
```
Comment (micronAMZN) [inline content/docs/user-guide/scripting/lua/debugging-tutorial.md:15]:
```suggestion
1. Open Lua Editor from the **Tools** menu.  
```

File: content/docs/user-guide/scripting/lua/debugging-tutorial.md [hunk 7]
```diff
@@ -1,171 +1,144 @@
...
+    
+    {{< note >}}
 The class reference feature is active only for the default context and component entity scripts.
+{{< /note >}}
 
-   After you connect, you can pause the execution of a given script by setting breakpoints.
+    After you connect, you can pause the execution of a given script by setting breakpoints.
 
-1. In the Lua Editor toolbar, click the **Breakpoints** icon ![Image NOT FOUND](/images/user-guide/lua-editor-debugger-breakpoints-icon.png) to show the **Breakpoints** window.
+1. In the Lua Editor toolbar, click the **Breakpoints** icon ![Breakpoints Icon](/images/user-guide/scripting/lua/lua-editor-debugger-breakpoints-icon.png) to show the **Breakpoints** window.
```
Comment (micronAMZN) [inline content/docs/user-guide/scripting/lua/debugging-tutorial.md:105]: No need to do this now, but now that we have svg icons in static/images/icons, we should look to get svg versions of these icon used on this page added to the directory and use the new icon shortcode.

File: content/docs/user-guide/scripting/lua/debugging-tutorial.md [hunk 8]
```diff
@@ -1,171 +1,144 @@
...
-1. Click the **Lua Locals** icon ![Image NOT FOUND](/images/user-guide/lua-editor-debugger-lua-locals-icon.png) to show local Lua variables.
+1. Click the **Lua Locals** icon ![Lua Locals Icon](/images/user-guide/scripting/lua/lua-editor-debugger-lua-locals-icon.png) to show local Lua variables.
 
-1. Click **Watched Variables** icon ![Image NOT FOUND](/images/user-guide/lua-editor-debugger-watched-variables-icon.png) to open the **Watched Variables** window, where you can specify variables to watch.
+1. Click **Watched Variables** icon ![Watched Variables Icon](/images/user-guide/scripting/lua/lua-editor-debugger-watched-variables-icon.png) to open the **Watched Variables** window, where you can specify variables to watch.
 
 1. Press **F11** a few times to step through the code. Note how the contents of the **Stack**, **Lua Locals**, and **Watched Variables** windows change.
+
 **Tip**
 For greater convenience, you can float or dock these windows.
```
Comment (micronAMZN) [inline content/docs/user-guide/scripting/lua/debugging-tutorial.md:124]:
```suggestion
{{< note >}}
For greater convenience, you can float or dock these windows.
{{< /note >}}
```

File: content/docs/user-guide/scripting/lua/debugging-tutorial.md [hunk 9]
```diff
@@ -1,171 +1,144 @@
...
+| ![Run in Editor Icon](/images/user-guide/scripting/lua/lua-editor-debugger-run-in-editor.png) | Run in Editor | Alt+F5 | Run in O3DE Editor. |
+| ![Run on Target Icon](/images/user-guide/scripting/lua/lua-editor-debugger-run-on-target.png) | Run on Target | Ctrl+F5 | Send script to the connected target and run it. |
+| ![Run/Continue Icon](/images/user-guide/scripting/lua/lua-editor-debugger-run-continue.png) | Run/Continue | F5 | Run or continue running the current script. |
+| ![Step Into Icon](/images/user-guide/scripting/lua/lua-editor-debugger-step-into.png) | Step Into | F11 | Step into the function called on the current line. |
+| ![Step Out Icon](/images/user-guide/scripting/lua/lua-editor-debugger-step-out.png) | Step Out | Shift+F11 | Step out of the called function. |
+| ![Step Over Icon](/images/user-guide/scripting/lua/lua-editor-debugger-step-over.png) | Step Over | F10 | Step over the function called on the current line. |
+| ![Toggle Breakpoint Icon](/images/user-guide/scripting/lua/lua-editor-debugger-toggle-breakpoint.png) | Toggle Breakpoint | F9 | Enable or disable a breakpoint on the current line. |
```
Comment (micronAMZN) [inline content/docs/user-guide/scripting/lua/debugging-tutorial.md:144]:
```suggestion
| ![Run in Editor Icon](/images/user-guide/scripting/lua/lua-editor-debugger-run-in-editor.png) | Run in Editor | **Alt+F5** | Run in O3DE Editor. |
| ![Run on Target Icon](/images/user-guide/scripting/lua/lua-editor-debugger-run-on-target.png) | Run on Target | **Ctrl+F5** | Send script to the connected target and run it. |
| ![Run/Continue Icon](/images/user-guide/scripting/lua/lua-editor-debugger-run-continue.png) | Run/Continue | **F5** | Run or continue running the current script. |
| ![Step Into Icon](/images/user-guide/scripting/lua/lua-editor-debugger-step-into.png) | Step Into | **F11** | Step into the function called on the current line. |
| ![Step Out Icon](/images/user-guide/scripting/lua/lua-editor-debugger-step-out.png) | Step Out | **Shift+F11** | Step out of the called function. |
| ![Step Over Icon](/images/user-guide/scripting/lua/lua-editor-debugger-step-over.png) | Step Over | **F10** | Step over the function called on the current line. |
| ![Toggle Breakpoint Icon](/images/user-guide/scripting/lua/lua-editor-debugger-toggle-breakpoint.png) | Toggle Breakpoint | **F9** | Enable or disable a breakpoint on the current line. |
```

File: content/docs/user-guide/scripting/lua/ebus.md [hunk 10]
```diff
@@ -14,11 +16,11 @@ Keep in mind the following points regarding the order of activation of Lua compo
...
+When a Lua script creates a handler object, it notifies a component attached to an entity that it should call the script handler functions when certain events occur. For example, in the first sample below, the script creates a spawner notification bus handler when `OnActivate()` is called. This tells the **Spawner** component attached to the entity that has the script to call the `OnSpawnBegin()`, `OnSpawnEnd()`, and `OnEntitySpawned()` functions when the spawner instantiates a new dynamic slice. Subsequently, the handler is explicitly disconnected and set back to nil in the `OnDeactivate` function. This ensures that processing time is not wasted when the entity attached to the script isn't active. As long as the entity is active, these functions are called by the Spawner component at the appropriate time.
```
Comment (micronAMZN) [inline content/docs/user-guide/scripting/lua/ebus.md:19]: Dynamic slices have been deprecated in favor of spawnables.

```suggestion
When a Lua script creates a handler object, it notifies a component attached to an entity that it should call the script handler functions when certain events occur. For example, in the first sample below, the script creates a spawner notification bus handler when `OnActivate()` is called. This tells the **Spawner** component attached to the entity that has the script to call the `OnSpawnBegin()`, `OnSpawnEnd()`, and `OnEntitySpawned()` functions when the spawner instantiates a new spawnable. Subsequently, the handler is explicitly disconnected and set back to nil in the `OnDeactivate` function. This ensures that processing time is not wasted when the entity attached to the script isn't active. As long as the entity is active, these functions are called by the Spawner component at the appropriate time.
```

**Review by micronAMZN — CHANGES_REQUESTED** [review]

File: content/docs/user-guide/scripting/lua/debugging-tutorial.md (hunk 7 above)
Comment (FiniteStateGit) [reply content/docs/user-guide/scripting/lua/debugging-tutorial.md:105]: Ok, I'll make a pass to replace with svg icons.

File: content/docs/user-guide/scripting/lua/lua-editor.md (hunk 1 above)
Comment (FiniteStateGit) [reply content/docs/user-guide/scripting/lua/lua-editor.md:39]: It is **Ctrl+Shift+]**, I will edit.

**Review by micronAMZN — APPROVED** [review]

**Review by willihay — CHANGES_REQUESTED** [review]
Lots of great improvements here! Just a few suggestions to address.

**Comment by FiniteStateGit** [comment]
Thank you for the suggestions and comments! I'll re-request review after I've pushed changes to headers and titles.

**Comment by FiniteStateGit** [comment]
Is **Console** treated as a tool?

**Comment by willihay** [comment]
> Is **Console** treated as a tool?

Debatable, but that seems to be the trend - at least when you're talking about the Console in O3DE Editor. The latest docs capitalize Console, for example **Editor Console**.

**Review by willihay — APPROVED** [review]

**Pull request merged**